### PR TITLE
build: upgrade builder to go1.6.3

### PR DESCRIPTION
**Pull request by petermattis**
And support for mapping in backtrace tools if available.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7984)

### CODE REVIEW

**Comment by petermattis** [comment]
This allows running the `coroner` and `hydra` tools within `builder.sh` if you have access to the private repo containing the associated binaries.

**Comment by BramGruneir** [comment]
LGTM

File: build/builder.sh [hunk 1]
```diff
@@ -76,16 +76,24 @@ cockroach_toplevel="$(dirname $(cd $(dirname $0); pwd))"
...
+vols="${vols} --volume=${gopath0}/pkg/docker_amd64:/usr/local/go/pkg/linux_amd64"
+vols="${vols} --volume=${gopath0}/pkg/docker_amd64_race:/usr/local/go/pkg/linux_amd64_race"
+vols="${vols} --volume=${gopath0}/bin/docker_amd64:/go/bin"
+vols="${vols} --volume=${HOME}/.jspm:/root/.jspm"
+vols="${vols} --volume=${HOME}/.npm:/root/.npm"
+vols="${vols} --volume=${cockroach_toplevel}:/go/src/github.com/cockroachdb/cockroach"
+
+backtrace_dir="${cockroach_toplevel}/../../cockroachlabs/backtrace"
+if test -d "${backtrace_dir}"; then
+  vols="${vols} --volume=${backtrace_dir}:/opt/backtrace"
```
Comment (tschottdorf) [inline build/builder.sh]: won't work if ${backtrace_dir} contains whitespace. Probably ok for now.

Comment (petermattis) [inline build/builder.sh]: Yeah, dealing with whitespace in filenames is such a pain. I'm going to leave as-is until someone complains.

**Pull request merged**